src/Pages/AdminTaskTable/components/ViewPayloadModal/ViewPayloadModal.tsx

```diff
@@ -0,0 +1,165 @@
+}
+
+const ViewPayloadModal = ({ adminTask, open, isFetching, setClosed }: ViewPayloadProps) => {
+  const classes = useStyles();
+
+  const [activeTabKey, setActiveTabKey] = useState<string | number>(0);
+  const detailRef = createRef<HTMLElement>();
+
+  const handleTabClick = (_, tabIndex: string | number) => {
+    setActiveTabKey(tabIndex);
+  };
+
+  const tabs = useMemo(() => {
+    const tabs: TabData[] = [];
+
+    if (!adminTask) {
+      return tabs;
+    }
+
+    tabs.push({
+      title: 'Payload',
+      data: adminTask.payload,
+      contentRef: createRef<HTMLElement>(),
+    });
+
+    switch (adminTask.typename) {
+      case 'snapshot': {
+        const pulp = adminTask.pulp;
+        if (pulp) {
+          if (pulp.sync) {
+            tabs.push({
+              title: 'Sync',
+              data: pulp.sync,
+              contentRef: createRef<HTMLElement>(),
+            });
+          }
+          if (pulp.distribution) {
+            tabs.push({
+              title: 'Distribution',
+              data: pulp.distribution,
+              contentRef: createRef<HTMLElement>(),
+            });
+          }
+          if (pulp.publication) {
+            tabs.push({
+              title: 'Publication',
+              data: pulp.publication,
+              contentRef: createRef<HTMLElement>(),
+            });
+          }
+        }
+        break;
+      }
+    }
+    return tabs;
+  }, [adminTask?.uuid]);
+
+  const actionTakingPlace = isFetching;
+
+  const isLoading = !adminTask || isFetching;
+
+  if (!open) return <></>;
+
+  return (
+    <Modal
+      position='top'
+      variant={ModalVariant.medium}
+      ouiaId='task_details'
+      ouiaSafe={!actionTakingPlace}
+      aria-label='Task details'
+      isOpen={open}
+      onClose={() => {
+        setActiveTabKey(0);
+        setClosed();
+      }}
+      header={
+        <Tabs
+          activeKey={activeTabKey}
+          onSelect={handleTabClick}
+          aria-label='Task tabs'
+          ouiaId='task-tabs'
+        >
+          <Tab
+            title={<TabTitleText>Task details</TabTitleText>}
+            aria-label='Task details'
+            ouiaId='task-details'
+            eventKey={0}
+            tabContentRef={detailRef}
+          />
+          {tabs.map(({ title, contentRef }) => (
+            <Tab
+              key={title}
+              eventKey={title}
+              aria-label={title}
+              ouiaId={title}
+              tabContentRef={contentRef}
+              title={<TabTitleText>{title}</TabTitleText>}
+            />
+          ))}
+        </Tabs>
+      }
+    >
+      <Hide hide={!isLoading}>
+        <Bullseye>
+          <Spinner />
+        </Bullseye>
+      </Hide>
+      <Hide hide={isLoading}>
+        <TabContent aria-label='Task details' eventKey={0} id='task-details' ref={detailRef}>
+          {/* Hide "isLoading" checks if adminTask is not null */}
+          <AdminTaskInfo adminTask={adminTask as AdminTask} />
```

Comment:
So if you look at the list tables, we have built out a [skeleton](https://www.patternfly.org/v4/components/skeleton) loader for those that roughly matches what will be shown on the screen. 
You should consider adding the same for this component.
We do this to "hopefully" prevent jittering modal sizes.

Reply:
![Screen Shot 2023-06-30 at 2 51 22 PM](https://github.com/content-services/content-sources-frontend/assets/38083295/296ee918-511f-4f01-b913-365b0976217a)

Further, you may want to have the tabs within the modal also be in a disabled state (if those tabs are guaranteed to be there).  If the "payload" tab may not be there, then a skeleton for that as well makes sense.